functions/go/set-annotations/annotations_transformer.go

```diff
@@ -43,15 +62,51 @@ func (p *plugin) Config(
 	return nil
 }
 
+// setEntry tracks mutated fields in Results and calls filtersutil.SetEntry
+func (p *plugin) setEntry(key, value, tag string) filtersutil.SetFn {
+	baseSetEntry := filtersutil.SetEntry(key, value, tag)
+	return func(node *kyaml.RNode) error {
+		p.Results = append(p.Results, &Result{
+			Value: value,
+			FieldPath: strings.Join(append(node.FieldPath(), key), "."),
+			FilePath: p.filePath,
+		})
+		return baseSetEntry(node)
+	}
+}
+
+// Filter implements the kio.Filter interface to update annotations using setEntry
+func (p *plugin) Filter(nodes []*kyaml.RNode) ([]*kyaml.RNode, error) {
```

Comment:
So you are forking the logic from upstream kyaml. I was suggesting to make the changes to upstream kyaml itself and get the results. https://github.com/kubernetes-sigs/kustomize/blob/master/api/filters/annotations/annotations.go#L26

Reply:
It seems there may have been a misunderstanding following our offline discussion of this feature. I made the assumption that I did not have the flexibility to adjust the signature of the shared/upstream kyaml library for the scope of this feature. After our discussion, it sounds like the right path forward is to adjust the signature and implement this functionality in the shared library. I will close this PR as I work on the alternative implementation.